src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -989,14 +1013,111 @@ private T getDataObjectInstanceFromResult(FindDataObjectsResponse.Entry e) {
             return (T) dataObject;
         }
 
-
         @Override
         public Iterator<T> iterator() {
             return new ResultIterator();
         }
     }
 
     /**
+     * The page subset of data objects that matched a {@code findDataObjects} query.
+     *
+     * @param <T> data object class to be returned
+     */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class SearchPage<T extends DXDataObject> implements Iterable<T> {
+        private final String classConstraint;
+        private final DXEnvironment env;
+        private final FindDataObjectsResponse response;
+
+        /**
+         * Iterator implementation for findDataObjects results page items.
+         */
+        @VisibleForTesting
+        private class ResultIterator implements Iterator<T> {
+
+            int lastElementIndex;
+
+            private ResultIterator() {
+                this.lastElementIndex = -1;
+            }
+
+            @Override
+            public boolean hasNext() {
+                return lastElementIndex < (response.results.size() - 1);
```

Comment:
Minor: seems a bit more natural to name the counter nextElementIndex, start it from 0, and remove the "-1" from the condition here.